docs/spec/spec.md

```diff
@@ -0,0 +1,1294 @@
+# Specification: Ballerina SerDes Library
+
+_Owners_: @MohamedSabthar @shafreenAnfar @ThisaruGuruge  
+_Reviewers_:   
+_Created_: 2022/08/01  
+_Updated_: 2022/08/01  
+_Edition_: Swan Lake  
+
+## Introduction
+This is the specification for the SerDes standard library of [Ballerina language](https://ballerina.io/), which is used for serializing and deserializing subtypes of Ballerina anydata type.
+
+The SerDes library specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant GitHub tag.
+
+If you have any feedback or suggestions about the library, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-standard-library/issues) or in the [Slack channel](https://ballerina.io/community/). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Proto3Schema](#2-proto3schema)
+	* 2.1 [`init` function](#21-init-function)
+	* 2.2 [`serialize` function](#22-serialize-function)
+	* 2.3 [`deserialize` function](#23-deserialize-function)
+3. [Ballerina anydata to proto3 mapping](#3-ballerina-anydata-to-proto3-mapping)
+	* 3.1 [Ballerina primitives](#31-ballerina-primitives)
+	* 3.2 [Array](#32-array)
+	* 3.3 [Union](#33-union)
+	* 3.4 [Record](#34-record)
+	* 3.5 [Map](#35-map)
+	* 3.6 [Table](#36-table)
+	* 3.7 [Tuple](#37-tuple)
+	* 3.8 [Enum](#38-enum)
+4. [Sample](#4-sample)
+
+## 1. Overview
+This specification elaborates on functionalities provided by SerDes library and how SerDes library maps the Ballerina anydata to a protocol buffer type.
+
+## 2. Proto3Schema
+`Proto3Schema` class provides APIs to perform serialization and deserialization of Ballerina anydata using protocol buffers (proto3). Following is the class definition of `Proto3Schema`
+
+```ballerina
+public class Proto3Schema {
+  *Schema;
```

Comment:
Let's add a section to describe the `Schema` class, and how someone can use it to write their own serializers/deserializers.